### PR TITLE
[ROCm] enable nvfuser

### PR DESCRIPTION
**Pull request by jeffdaily**
### Description
The nvfuser is enabled for ROCm.

### Testing
CI label ciflow/trunk covers the newly enabled ROCm functionality as well as any CUDA regressions caused by these changes.

### CODE REVIEW

**Comment by davidberard98** [comment]
@jeffdaily can you run with slow tests? `PYTORCH_TEST_WITH_SLOW=1` will add these test.

I also see that some other NVFuser tests are disabled, (search NVFuser on the rocm (1, 2) CI job)

**Comment by jeffdaily** [comment]
@davidberard98 added the ciflow/periodic label since this is where we run the slow tests for ROCm.  It revealed an unexpected success.

```
2022-08-29T17:48:25.7390645Z   test_nvfuser_extremal_values_unique_consecutive_cuda_float16 (__main__.TestCudaFuserOpInfoCUDA) ... unexpected success (6.622s)
2022-08-29T17:48:32.3703269Z   test_nvfuser_extremal_values_unique_consecutive_cuda_float32 (__main__.TestCudaFuserOpInfoCUDA) ... unexpected success (6.631s)
2022-08-29T17:48:38.9635277Z   test_nvfuser_extremal_values_unique_consecutive_cuda_float64 (__main__.TestCudaFuserOpInfoCUDA) ... unexpected success (6.593s)
2022-08-29T17:48:54.6667431Z   test_nvfuser_extremal_values_unique_cuda_float16 (__main__.TestCudaFuserOpInfoCUDA) ... unexpected success (15.703s)
2022-08-29T17:49:10.5367634Z   test_nvfuser_extremal_values_unique_cuda_float32 (__main__.TestCudaFuserOpInfoCUDA) ... unexpected success (15.870s)
2022-08-29T17:49:26.3813820Z   test_nvfuser_extremal_values_unique_cuda_float64 (__main__.TestCudaFuserOpInfoCUDA) ... unexpected success (15.845s)
```

Should be fixed by https://github.com/pytorch/pytorch/pull/82498/commits/6bbe7a79fbfd45b8b0b0ecfa7cdabc0822558e19.

Unskipped the few NVFuser tests you found in the logs.  See https://github.com/pytorch/pytorch/pull/82498/commits/47da32aac994e9f43b2fade86d431a80e37a19e0.  They passed.  So now just waiting on another round of CI.

**Comment by jeffdaily** [comment]
@davidberard98 I hope the phabricator build is going well.  All CI is green, [including rocm slow](https://github.com/pytorch/pytorch/runs/8079457676?check_suite_focus=true).

File: torch/csrc/jit/codegen/cuda/nvfuser.cmake
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/nvfuser.cmake:12]: nitpick. Can we guard rocm only files under `if(USE_ROCM)`?

**Review by jjsjann123 — APPROVED** [review]
Looks mostly just mechanical changes. went through most files except the rocm specific runtime headers.

A nitpick on the cmake file to guard rocm header under `USE_ROCM`. not sure if there's similar thing we can do with `build_variables.bzl`

File: torch/csrc/jit/codegen/cuda/nvfuser.cmake
Comment (jeffdaily) [reply torch/csrc/jit/codegen/cuda/nvfuser.cmake:12]: What about the [bzl file](https://github.com/pytorch/pytorch/blob/6bbe7a79fbfd45b8b0b0ecfa7cdabc0822558e19/build_variables.bzl)?  

The overhead of keeping these four rocm-specific files in the NVFUSER_RUNTIME_FILES cmake var is minimal; four extra files to stringify at build time.  I know it is a nit, but I'd rather avoid yet another commit to this PR since we're in a  good state at the moment.

Comment (davidberard98) [reply torch/csrc/jit/codegen/cuda/nvfuser.cmake:12]: I'll accept & land this, and @jeffdaily can you open another PR to address Jie's comments once this lands?

**Review by davidberard98 — APPROVED** [review]

**Comment by davidberard98** [comment]
@jeffdaily looks good, let me know if you want to wait on anything before landing - otherwise I'll land later today (or feel free to land it yourself).

File: torch/csrc/jit/codegen/cuda/nvfuser.cmake
Comment (jeffdaily) [reply torch/csrc/jit/codegen/cuda/nvfuser.cmake:12]: See: https://github.com/pytorch/pytorch/pull/84312.

**Pull request closed without merging**
